packages/legend-query/src/stores/filterOperators/QueryBuilderFilterOperatorHelper.ts

```diff
@@ -443,22 +449,38 @@ export const buildFilterConditionState = (
   // This is used for post-build checks (useful when this expression is nested inside a longer
   // chain of expression, e.g. ->exists($x|x == 'something'), etc.)
   let mainExpressionWithOperator: SimpleFunctionExpression | undefined;
-
-  if (matchFunctionName(expression.functionName, operatorFunctionFullPath)) {
+  const reservedDateTimeOperatorFunctionFullPath = Array.from(
```

Comment:
Same as Mauricio, I think we shouldn't expose those function at this level. This is an utility and when we introduce `QUERY_BUILDER_RESERVED_DATETIME_OPERATOR_FUNCTION_TO_LABEL` here, we are adding unecessary logic which could have been added to the operator function itself. When you put this logic here, we also lose some nuances of the logic: we know for example, `IS_ON_DAY` only applies to `EQUAL` operator.